traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go

```diff
@@ -1194,25 +1195,28 @@ WHERE ds.id=$1
 	cdnDomain := ""
 	cdnName := ""
 	if err := tx.QueryRow(q, id).Scan(&xmlID, &protocol, &dsTypeStr, &routingName, &cdnDomain, &cdnName); err != nil {
-		return "", "", "", fmt.Errorf("querying delivery service %v host name: "+err.Error()+"\n", id)
+		if err == sql.ErrNoRows {
+			return "", "", "", fmt.Errorf("querying delivery service %v host name: "+err.Error(), id), http.StatusNotFound
+		}
```

Comment:
Once this error is user-visible, the user should not see any database-related error like `no rows in result set`.